### PR TITLE
Replace encoding/json with jsoniter.

### PR DESCRIPTION
**Pull request by fasaxc**
## Description

Using jsoniter's drop-in replacement mode to speed up our JSON encoding/decoding.  Seems to be a simple win; cuts the time to decode a workload endpoint in half, for example.  This has a significant impact on parsing the Typha protocol.

## Todos

- [ ] Tests
- [ ] Documentation
- [ ] Release note

## Release Note

```release-note
Calico now uses a faster JSON parsing library; this reduces CPU load and improves start-up latency.
```

## Reminder for the reviewer

Make sure that this PR has the correct labels and milestone set.

Every PR needs one `docs-*` label.

- `docs-pr-required`: This change requires a change to the documentation that has not been completed yet.
- `docs-completed`: This change has all necessary documentation completed.
- `docs-not-required`: This change has no user-facing impact and requires no docs.

Every PR needs one `release-note-*` label.

- `release-note-required`: This PR has user-facing changes. Most PRs should have this label.
- `release-note-not-required`: This PR has no user-facing changes.

Other optional labels:

- `cherry-pick-candidate`: This PR should be cherry-picked to an earlier release. For bug fixes only.
- `needs-operator-pr`: This PR is related to install and requires a corresponding change to the operator.

### CODE REVIEW

File: libcalico-go/lib/net/mac.go
Comment (caseydavenport) [inline libcalico-go/lib/net/mac.go:30]: Would it make sense to have a libcalico-go package to wrap this up?

```go
package json

jsoniter "github.com/json-iterator/go"


func Marshal(s string) []byte, error {
  return jsoniter.ConfigCompatibleWithStandardLibrary.Marshal(s)
}
```

Would turn this into an import change rather than an import + local variable change. Unless you think there's a future where we want to use different jsoniter implementations in different places?

**Comment by marvin-tigera** [comment]
OK, I will merge the pull request when it's ready, leave the commits as is when I merge it, and leave the branch after I've merged it.

**Review by caseydavenport — APPROVED** [review]

**Pull request merged**
